backend/src/util.py

```diff
@@ -14,3 +16,28 @@ def wrapper():
         return result, duration
 
     return wrapper
+
+
+###
+### ChatGPT wrote the below code. I have no idea how it works, but it does.
+###
+
+
+def combine_sets(sets: list[set[NodeId]]) -> list[set[NodeId]]:
```

Comment:
Yes that is correct. I tried to write an algorithm for this myself and couldn't wrangle something together. I can try again today but I make no guarantees that it won't end up being slower, worse, messier, etc than this.

Reply:
re point 2: I don't think its possible for this to be any better than O(n^2). How would you write this in a way that doesn't compare every item in every set with every item in every other set?

Reply:
I've got an idea for a different algorithm that might be able to do it all with just 2 for loops and would only be On^2 in the worst case

Reply:
So my idea didn't totally work. I'd have to do multiple passes, meaning that it would require a while loop and going over it multiple times until everything is merged. Even without that, my code was slower than the implementation here. So, I've decided to just make it copy the sets so it isn't modifying the variable, and leave the function.

If you can come up with a better algorithm, be my guest. But I cannot.